File: DSA/fundamentals/arrays_and_lists/part1/python/judge.py
from solution import Solution

class Judge:
    def run_tests(self):
        test_cases_p = [
            [1, 2, 4, 5],
            [1, 2, 4, 5],
        ]
        test_cases_m = [
            [2, 2, 5, 10],
            [2, 1, 5, 0],
        ]

        passes = 0;
        for i in range(len(test_cases_p)):
            if self.test(test_cases_p[i].copy(), test_cases_m[i].copy(), self.test_case_solution_output(test_cases_p[i].copy(), test_cases_m[i].copy())):
                passes += 1
        print(f"Passed {passes} of {len(test_cases_p)} tests.")

    def test(self, p, m, expected):
        solution = Solution()
        output = solution.difference(p, m)
        passed = output == expected

        print(f"p: {p}")
        print(f"m: {m}")
        print(f"Expected Output: {expected}")
        print(f"Your Output: {output}")
        print("Test " + ("PASSED" if passed else "FAILED"))
        print("-----------")
        return passed

    def test_case_solution_output(self, p, m):
        sum = 0
        for i in range(len(p)):
            sum += m[i] - p[i]
        return sum

File: DSA/fundamentals/arrays_and_lists/part1/javascript/judge.js
const Solution = require('./solution');

class Judge {
    runTests() {
        const testCasesP = [
            [1, 2, 4, 5],
            [1, 2, 4, 5],
        ];
        const testCasesM = [
            [2, 2, 5, 10],
            [2, 1, 5, 0],
        ];

        let passes = 0;
        for (let i = 0; i < testCasesP.length; i++) {
            if (this.test(testCasesP[i], testCasesM[i], this.testCaseSolutionOutput(testCasesP[i], testCasesM[i]))) {
                passes++;
            }
        }
        console.log(`Passed ${passes} of ${testCasesP.length} tests.`);
    }

    test(p, m, expected) {
        const solution = new Solution();
        const output = solution.difference(p, m);
        const passed = output === expected;

        console.log(`p: ${JSON.stringify(p)}`);
        console.log(`m: ${JSON.stringify(m)}`);
        console.log(`Expected Output: ${expected}`);
        console.log(`Your Output: ${output}`);
        console.log(passed ? "Test PASSED" : "Test FAILED");
        console.log("-----------");
        return passed;
    }

    testCaseSolutionOutput(p, m) {
        let sum = 0;
        for (let i = 0; i < p.length; i++) {
            sum += m[i] - p[i];
        }
        return sum;
    }
}

module.exports = Judge;